aws/data_source_aws_sfn_activity.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sfn"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsSfnActivity() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsSfnActivityRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"creation_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsSfnActivityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient)
+	conn := client.sfnconn
+	log.Print("[DEBUG] Reading Step Function Activity")
+
+	params := &sfn.DescribeActivityInput{}
+
+	var name string
+	if nm, ok := d.GetOk("name"); ok {
+		name = nm.(string)
+		params.ActivityArn = aws.String(buildSfnActivityArn(name, client.accountid, client.region))
```

Comment:
Manually generating ARN values can be problematic as the account ID may not be present in the provider initialization (see the [`skip_requesting_account_id` argument](https://www.terraform.io/docs/providers/aws/index.html#skip_requesting_account_id)) and this function currently assumes AWS Commercial for the ARN partition (`arn:aws:`). Please see https://github.com/terraform-providers/terraform-provider-aws/pull/10932 for an example of using a paginated List function ([`ListActivitiesPages()`](https://docs.aws.amazon.com/sdk-for-go/api/service/sfn/#SFN.ListActivitiesPages) in this case) to lookup the ARN via name.